### ISSUE TITLE
Missing active_low attribute

### ISSUE THREAD

**Issue by sr-murthy** [derekstavis/python-sysfs-gpio#4]
Hi

I am currently working with a GPIO chip controller (the chip is of type 'ICH', sorry I don't know much about the hardware) - one of the attributes of pins on this chip is `active_low`, which determines whether the logic of activating the pin is inverted, i.e. if `active_low = 0` then basically for pin values 1 = high and 0 = low, but if `active_low = 1` then this is inverted to become 1 = low and 0 = high.

I cannot see this being indicated in the `Pin` class constructor or the `Controller` class you have written.  I would like to add this via a PR.  Is that OK?

SM

**Comment by derekstavis** [comment]
Hey SM,

First, I'm glad to see that you are using this package!

Indeed `active_low` isn't actually implemented. Although I haven't noticed this on GPIO controllers I used when implementing the module, [`active_low` is a Sysfs GPIO standard](https://www.kernel.org/doc/Documentation/gpio/sysfs.txt), and I would be very glad to accept a PR implementing it! Go for it! :+1: 

Thanks!
Derek

**Comment by sr-murthy** [comment]
Thanks for the info.  Actually I wrote my own API for the sysfs GPIO before I came across this, and noticed there were some differences.  It seems the way you read and write to the sysfs GPIO filesis via Python file objects, but I found that this was not actually not possible with my setup (I was running Ubuntu 15.10 and the GPIO driver was of type `ich`) because for example `/sys/class/gpio/export` and `/sys/class/gpio/unexport` could not be read or written to, even as `root`.  So I had to use system calls via `subprocess`.  So for example, here's some of my code for exporting a pin, writing a pin value (assuming it is a 'out' pin), and listing exported pins as integers.
```
from subprocess import CalledProcessError, check_output, STDOUT
....
# export pin
try:
    subprocess.call('echo %d > /sys/class/gpio/export' % pin_number, stderr=STDOUT, shell=True)
except CalledProcessError
    raise

# set pin value
try:
    call('echo %d > /sys/class/gpio/gpio%d/value' % (value, pin_number), stderr=STDOUT, shell=True)
except CalledProcessError:
    raise

# list exported pins as numbers
['gpio436,gpio462']
import re
try:
    map(
        lambda s: int(s.strip('gpio')),
        filter(
            lambda t: re.match(r'gpio[0-9]{3}$', t),
            check_output('ls /sys/class/gpio', stderr=STDOUT, shell=True).strip('\n').split('\n')
        )
    )
except CalledProcessError:
    raise
```

**Comment by derekstavis** [comment]
Damn, this is really sad! The scenario I use is popular embedded devices, like Raspberry Pi, in which case the module works perfectly. I found strange that Python can't write to the files, as `echo 1 > /sys/class/gpio/gpioN/value` actually opens a subprocess (in this case user's shell) as root, and does pretty the same as what Python does: Opening a file and then writing to it.

**Comment by sr-murthy** [comment]
So that seems an issue then with the hardware perhaps?  What make is the GPIO chip you use with the Pi?  The ICH chip is Intel I believe.

http://cateee.net/lkddb/web-lkddb/GPIO_ICH.html

**Comment by sr-murthy** [comment]
I don't think it is a Python issue in my case - I could not for example do `cat /sys/class/gpio/export`, it would complain of an I/O error or some input error (I can't remember, I'm not at my desk).  These files that sysfs exposes are not real files, if they were you could `cat` them.

**Comment by derekstavis** [comment]
Solved by #5!
